src/OmnichannelChatSDK.ts

```diff
@@ -1693,6 +1635,118 @@ class OmnichannelChatSDK {
         }
     }
 
+    public async getAgentAvailability(optionalParams: GetAgentAvailabilityOptionalParams = {}): Promise<any> { // eslint-disable-line @typescript-eslint/no-explicit-any
+        const reportError = (response: string, message: string, chatId = "") => {
+            const exceptionDetails: ChatSDKExceptionDetails = {
+                response,
+                message
+            };
+
+            this.scenarioMarker.failScenario(TelemetryEvent.GetAgentAvailability, {
+                RequestId: this.requestId,
+                ExceptionDetails: JSON.stringify(exceptionDetails),
+                ChatId: chatId
+            });
+
+            throw new Error(exceptionDetails.message);
+        }
+
+        this.scenarioMarker.startScenario(TelemetryEvent.GetAgentAvailability, {
+            RequestId: this.requestId
+        });
+
+        if (!this.authSettings) {
+            reportError("UnAuthUnsupported", "GetAgentAvailability is supported only for authenticated live chat widget.");
```

Comment:
Would `Unsupported` be better? We'll be able to reuse this type of response in other scenarios. We can be more descriptive in the message